js/repl/UriUtils.js

```diff
@@ -84,7 +85,9 @@ const updateQuery = (state: PersistedState) => {
       return null;
     } else if (key === "code") {
       return `${key}_lz=` + compress(state.code);
-    } else {
+      // Default state can be empty string or a boolean value
+      // URL State should not be empty string and default state
+    } else if(state[key] !== "" && state[key] !== persistedStateDefaults[key]){
```

Comment:
I raised concerns about this initially. Removing values from the URL could result in a shared link behaving differently between 2 users b'c of `localStorage` values (see my comment below). If we remove values from the URL (eg your changes in `UriUtils`) we need to also change the override/restore behavior elsewhere. I think for now, we should revert the changes to `UriUtils`.